### PR TITLE
Solution Wide: Fix Warnings

### PR DESCRIPTION
**Pull request by IsaacMarovitz**
+ Fixes several warnings in `Ryujinx.Ava`
+ Removes unnecessary casts and type specifiers solution-wide (majority of edits)
+ Sets enums used with bitwise operators as `[Flags]`

 Warnings changed as suggested by JetBrains Rider. This may cause unintended errors as Rider does not completely understand Avalonia.

Results in an overall reduction in warnings of ~1-2K

**Please test thoroughly**

Should not affect functionality except when occasional oversights/bugs are fixed.

### CODE REVIEW

File: Ryujinx.Ava/Ui/Controls/GameGridView.axaml.cs [hunk 1]
```diff
@@ -54,12 +52,7 @@ private void InitializeComponent()
         {
             AvaloniaXamlLoader.Load(this);
         }
-
-        private void SearchBox_OnKeyUp(object sender, KeyEventArgs e)
-        {
-            (DataContext as MainWindowViewModel).SearchText = (sender as TextBox).Text;
-        }
-
```
Comment (TSRBerry) [inline Ryujinx.Ava/Ui/Controls/GameGridView.axaml.cs:62]: This is still needed.

File: Ryujinx.Ava/Ui/Controls/GameListView.axaml.cs [hunk 2]
```diff
@@ -54,12 +52,7 @@ private void InitializeComponent()
         {
             AvaloniaXamlLoader.Load(this);
         }
-
-        private void SearchBox_OnKeyUp(object sender, KeyEventArgs e)
-        {
-            (DataContext as MainWindowViewModel).SearchText = (sender as TextBox).Text;
-        }
-
```
Comment (TSRBerry) [inline Ryujinx.Ava/Ui/Controls/GameListView.axaml.cs:62]: This too.

**Review by TSRBerry — COMMENTED** [review]
Other than the ones I commented on, I wasn't able to find more unintentional deletions, so this should be fine otherwise.

**Comment by TSRBerry** [comment]
You removed a few "explicitly typed new"s that I think would be better to keep for readability.
But we don't have a rule / standard for that  yet, so I'm not sure how to handle this.

**Comment by IsaacMarovitz** [comment]
> You removed a few "explicitly typed new"s that I think would be better to keep for readability. But we don't have a rule / standard for that yet, so I'm not sure how to handle this.

In that case, we should adjust the config to ignore those errors. Otherwise, errors lose their meaning.

**Comment by TSRBerry** [comment]
That's why I'm saying we currently don't have a style standard for this. There was a discussion about this before, but no conclusion was reached.

File: Ryujinx.Tests/Cpu/CpuTestSimd.cs [hunk 3]
```diff
@@ -82,7 +82,7 @@ public sealed class CpuTestSimd : CpuTest
 #region "ValueSource (Types)"
         private static ulong[] _1B1H1S1D_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x000000000000007Ful,
+            return new[] { 0x0000000000000000ul, 0x000000000000007Ful,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimd.cs:85]: Those initializers are misaligned now.

File: Ryujinx.Tests/Cpu/CpuTestSimd32.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ private static uint[] _Vabs_Vneg_F_()
 #region "ValueSource (Types)"
         private static ulong[] _8B4H2S_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
+            return new[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimd32.cs:38]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdCvt.cs [hunk 5]
```diff
@@ -17,13 +17,13 @@ public sealed class CpuTestSimdCvt : CpuTest
 #region "ValueSource (Types)"
         private static uint[] _W_()
         {
-            return new uint[] { 0x00000000u, 0x7FFFFFFFu,
+            return new[] { 0x00000000u, 0x7FFFFFFFu,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdCvt.cs:20]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdExt.cs [hunk 6]
```diff
@@ -14,7 +14,7 @@ public sealed class CpuTestSimdExt : CpuTest
 #region "ValueSource"
         private static ulong[] _8B_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
+            return new[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdExt.cs:17]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs [hunk 7]
```diff
@@ -28,7 +28,7 @@ private static uint[] _Vrint_AMNP_V_F32_()
 #region "ValueSource (Types)"
         private static uint[] _1S_()
         {
-            return new uint[] { 0x00000000u, 0x7FFFFFFFu,
+            return new[] { 0x00000000u, 0x7FFFFFFFu,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs:31]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdImm.cs [hunk 8]
```diff
@@ -50,13 +50,13 @@ private static byte ShrinkImm64(ulong imm64)
 #region "ValueSource (Types)"
         private static ulong[] _2S_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7FFFFFFF7FFFFFFFul,
+            return new[] { 0x0000000000000000ul, 0x7FFFFFFF7FFFFFFFul,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdImm.cs:53]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdIns.cs [hunk 9]
```diff
@@ -14,46 +14,46 @@ public sealed class CpuTestSimdIns : CpuTest
 #region "ValueSource"
         private static ulong[] _1D_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7FFFFFFFFFFFFFFFul,
+            return new[] { 0x0000000000000000ul, 0x7FFFFFFFFFFFFFFFul,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdIns.cs:17]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdLogical32.cs [hunk 10]
```diff
@@ -13,7 +13,7 @@ public sealed class CpuTestSimdLogical32 : CpuTest32
 #region "ValueSource (Types)"
         private static ulong[] _8B4H2S_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
+            return new[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdLogical32.cs:16]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdReg.cs [hunk 11]
```diff
@@ -16,7 +16,7 @@ public sealed class CpuTestSimdReg : CpuTest
 #region "ValueSource (Types)"
         private static ulong[] _1B1H1S1D_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x000000000000007Ful,
+            return new[] { 0x0000000000000000ul, 0x000000000000007Ful,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdReg.cs:19]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdReg32.cs [hunk 12]
```diff
@@ -126,14 +126,14 @@ private static uint[] _Vq_Add_Sub_I_()
 #region "ValueSource (Types)"
         private static ulong[] _8B1D_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
+            return new[] { 0x0000000000000000ul, 0x7F7F7F7F7F7F7F7Ful,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdReg32.cs:129]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdRegElem.cs [hunk 13]
```diff
@@ -14,21 +14,21 @@ public sealed class CpuTestSimdRegElem : CpuTest
 #region "ValueSource (Types)"
         private static ulong[] _2S_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7FFFFFFF7FFFFFFFul,
+            return new[] { 0x0000000000000000ul, 0x7FFFFFFF7FFFFFFFul,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdRegElem.cs:17]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdShImm.cs [hunk 14]
```diff
@@ -17,37 +17,37 @@ public sealed class CpuTestSimdShImm : CpuTest
 #region "ValueSource (Types)"
         private static ulong[] _1D_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7FFFFFFFFFFFFFFFul,
+            return new[] { 0x0000000000000000ul, 0x7FFFFFFFFFFFFFFFul,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdShImm.cs:20]: Misaligned.

File: Ryujinx.Tests/Cpu/CpuTestSimdShImm32.cs [hunk 15]
```diff
@@ -13,33 +13,33 @@ public sealed class CpuTestSimdShImm32 : CpuTest32
 #region "ValueSource (Types)"
         private static ulong[] _1D_()
         {
-            return new ulong[] { 0x0000000000000000ul, 0x7FFFFFFFFFFFFFFFul,
+            return new[] { 0x0000000000000000ul, 0x7FFFFFFFFFFFFFFFul,
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestSimdShImm32.cs:16]: Misaligned.

**Review by gdkchan — COMMENTED** [review]
Most of those are "suggestions", I'm not sure why Rider considers them warnings by default (?).

**Comment by AcK77** [comment]
I suggest you to PR things as small part to make reviewing and bug tracking easier.

**Pull request closed without merging**